llvm/lib/CodeGen/SelectionDAG/DAGCombiner.cpp

```diff
@@ -16320,8 +16320,8 @@ SDValue DAGCombiner::visitFADDForFMACombine(SDNode *N) {
   if (!HasFMAD && !HasFMA)
     return SDValue();
 
-  bool AllowFusionGlobally = (Options.AllowFPOpFusion == FPOpFusion::Fast ||
-                              Options.UnsafeFPMath || HasFMAD);
+  bool AllowFusionGlobally =
+      Options.AllowFPOpFusion == FPOpFusion::Fast || HasFMAD;
```

Comment:
There are other uses of `UnsafeFPMath` in this function, and in `visitFSUBForFMACombine`, and other functions related to FMA formation. Do you plan to remove more of them? Or at least all the ones that also test `AllowFPOpFusion`?

Reply:
Currently only drop `unsafe-fp-math` in global fusion and handle regressions, I will handle `visitFSUBForFMACombine` later. Eliminate other `UnsafeFPMath`s would be a long term goal.

Reply:
The fsub case is the next logical step here

Reply:
OK